lib/index.js

```diff
@@ -128,9 +128,10 @@ function HttpProxyMiddleware(context, opts) {
     }
 
     function logError(err, req, res) {
-        var hostname = (req.hostname || req.host) || (req.headers && req.headers.host); // (node0.10 || node 4/5) || (websocket)
-        var targetUri = (proxyOptions.target.host || proxyOptions.target) + req.url;
+        var hostname = (req.headers && req.headers.host) || (req.hostname || req.host);     // (websocket) || (node0.10 || node 4/5)
+        var target = proxyOptions.target.host || proxyOptions.target;
+        var errReference = 'https://nodejs.org/api/errors.html#errors_common_system_errors'; // link to Node Common Systems Errors page
 
-        logger.error('[HPM] PROXY ERROR: %s. %s -> %s', err.code, hostname, targetUri);
+        logger.error('[HPM] Error occurred while trying to proxy request %s from %s to %s (%s) \n(%s)', req.url, hostname, target, err.code, errReference);
```

Comment:
@chimurai good point. I was only thinking console output in create-react-app.  I looked there and line-break was used to break long lines.

I agree for the logging purpose we should remove the line-break.  I can update the pull request.   I understand with your onError suggestion, this pull request won't be necessary for create-react-app use case, but the error message does look a little better overall (plus Node error page link).
Also would you like to use [http-proxy-middleware] or keep [HPM]?